### PR TITLE
ANDROAPP-5644-mobile-ui-update-dropdown-component

### PR DESCRIPTION
**Pull request by msasikanth**
- Bring dropdown menu inside the component
- Show bottom sheet menu if there are more than 7 items
- Change input dropdown function signature

### CODE REVIEW

**Comment by msasikanth** [comment]
Designs were updated to include changes to dropdown menu, updating it.

**Review by xavimolloy — APPROVED** [review]

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDropDown.kt [hunk 1]
```diff
@@ -33,22 +52,138 @@ import org.hisp.dhis.mobile.ui.designsystem.theme.TextColor
...
+            },
+        )
+    }
+
+    if (dropdownItems.size > MAX_DROPDOWN_ITEMS) {
+        Box {
+            inputField(modifier)
+
+            if (showDropdown) {
+                BottomSheetShell(
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDropDown.kt:98]: Not sure, but should we need the searchbar to meet the requirements?
image

Comment (msasikanth) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDropDown.kt:98]: Good point, let me confirm that with Marcos and make that change if required.

Comment (msasikanth) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDropDown.kt:108]: Made the change

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDropDown.kt [hunk 2]
```diff
@@ -33,22 +54,165 @@ import org.hisp.dhis.mobile.ui.designsystem.theme.TextColor
...
+                    }
+                }
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun InputField(
```
Comment (xavimolloy) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDropDown.kt:202]: What do you think of renaming this class to something like DropdownInputField? we already have a file that's called InputField that contains the BasicTextField we use in most inputs

**Review by xavimolloy — APPROVED** [review]

**Review by xavimolloy — APPROVED** [review]

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDropDown.kt [hunk 3]
```diff
@@ -33,22 +54,165 @@ import org.hisp.dhis.mobile.ui.designsystem.theme.TextColor
...
+    selectedItem: DropdownItem? = null,
     supportingTextData: List<SupportingTextData>? = null,
     legendData: LegendData? = null,
     isRequiredField: Boolean = false,
     modifier: Modifier = Modifier,
     onFocusChanged: ((Boolean) -> Unit)? = null,
     onResetButtonClicked: () -> Unit,
-    onArrowDropDownButtonClicked: () -> Unit,
+    onItemSelected: (DropdownItem) -> Unit,
+    showSearchBar: Boolean = true,
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDropDown.kt:73]: is this optional? I thought it was always visible for dropdown in the bottombar case

Comment (msasikanth) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDropDown.kt:73]: Yep, Marcos wanted it to be configurable by the app whether to show the search or not.

**Review by andresmr — APPROVED** [review]

**Pull request merged**